### DISCUSSION TITLE
Closing an error popup whenever it is present in the DOM

### DISCUSSION THREAD

**Discussion by HeshamMoneer** [cypress-io/cypress#20115] [Questions and Help]
The problem is, sometimes an error popup is displayed in the page while a test is running. This does not happen if the test runs manually. I have not spotted the reason for that issue yet, but until then, is there a possibility to tell cypress to close that popup whenever it is open? If I intercept in the test manually and close the popup, the test continues and passes successfully.

**Comment by portyu9** [comment]
Hi @HeshamMoneer 

Yes. If the popup may or may not be present, check for it synchronously from the page body:

```js
cy.get('body').then(($body) => {
  const $popup = $body.find('[data-cy="error-popup"]')

  if ($popup.length) {
    cy.wrap($popup)
      .find('[data-cy="close"]')
      .click()
  }
})
```

I would put that into a custom command and call it after the action/request that can produce the popup.

Do not use:

```js
cy.get('[data-cy="error-popup"]')
```

as the conditional check, because Cypress will retry and eventually fail when the popup is absent.

One important caveat: this is reliable only after the page state has settled. If the popup appears asynchronously at an unpredictable time, the better fix is to identify/control the request or application state causing it rather than continuously watching the DOM.

Please mark this answer as accepted if it helped.
